src/Auth/Directives.js

```diff
@@ -108,11 +108,34 @@ class AuthenticatedDirective extends SchemaDirectiveVisitor {
   }
 }
 
+class OwnerDirective extends SchemaDirectiveVisitor {
 
+  visitObject(type){
+    this.wrapOwnerAuth(type);
+    type._requiresOwAuth = true;
+  }
+    
+  visitFieldDefinition(field) {
+    const { resolve = defaultFieldResolver } = field;
+    field.resolve = async function (record, args, context, info){
+
+      const OwnerRequester = await getOwnerid(context.token.owner);
+      const OwnerRequested = await getOwnerid(record);
+
+      if(OwnerRequester !== null && OwnerRequested !== null){
+        if(OwnerRequester == OwnerRequested){
```

Comment:
Always use the `===` when checking equality as this ensure that not only the values are the same but both must be of the same type.